ARCs/arc-0018.md

```diff
@@ -0,0 +1,601 @@
+
+#### Asset Offer
+
+The Asset Offer is the a data structure stored in the owners local state. It is keyed in local storage by the byte string representing the ASA Id. 
+
+```ts
+interface AssetOffer {
+    auth_address:   string // The address of a marketplace or account that may issue a transfer request
+    offered_amount: number // The number of units being offered
+}
+```
+
+This concept is important to this specification because we use the clawback feature to transfer the assets. Without some signal that the current owner is willing to have their assets transferred, it is be possible to transfer the asset without their permission. 
+
+In order for a transfer to occur, this field **MUST** be set and the parameters of the transfer request **MUST** match the value set.
+
+> A transfer matching the offer would require the transfer amount <= offered amount and that the transfer is sent by auth_address
+
+#### Royalty Asset Parameters 
+
+The Clawback parameter **MUST** be set to the Application Address of the Royalty Enforcer.
+
+> Since the Royalty Enforcer relies on using the Clawback mechanism to perform the transfer the Clawback should NEVER be set to the zero address. 
+
+The Freeze parameter **MUST** be set to the Application Address of the Royalty Enforcer if `FreezeAddr != ZeroAddress`, else set to `ZeroAddress`.
+
+> If the creator wants to allow an ASA to be Royalty Free after some conditions are met, it should be set to the Application Address 
+
+The Manager parameter **MUST** be set to the Application Address of the Royalty Enforcer if `ManagerAddr != ZeroAddress`, else set to `ZeroAddress`.
+
+> If the creator wants to update the Freeze parameter, this should be set to the application address
+
+The Reserve parameter **MAY** be set to anything.
+
+The `DefaultFrozen` **MUST** be set to true.
+
+### Third Party Marketplace
+
+In order to support secondary sales on external markets this spec was designed such that the Royalty Asset may be listed without transferring it from the current owners account.  
+
+A Marketplace may call the transfer request as long as the address initiating the transfer has been set as the `authorized_address` through the [offer](#offer) method in some previous transaction by the current owner.
+
+### ABI Methods
+
+The following is a set of methods that conform to the [ABI](arc-0004.md) specification meant to enable the configuration of a Royalty Policy and perform transfers.
+
+#### Set Policy  (REQUIRED):
+    
+```
+set_policy(royalty_basis: uint64, royalty_recipients: account)
+```
+
+Sets the policy for any assets using this application as a Royalty Enforcer.
+
+The `royalty_basis` is the percentage for royalty payment collection, specified in basis points (e.g., 1% is 100).  
+
+A Royalty Basis **MUST** be immutable, if an application call is made that would overwrite an existing value it **MUST** fail. 
```

Comment:
I thought we were not going to force this and should offer language to put a max in here